xchange-poloniex2/src/main/java/info/bitrich/xchangestream/poloniex2/dto/PoloniexWebSocketEventsTransaction.java

```diff
@@ -19,14 +20,37 @@
  */
 @JsonFormat(shape = JsonFormat.Shape.ARRAY)
 public class PoloniexWebSocketEventsTransaction {
-    public String channelId;
-    public String seqId;
-    public JsonNode[] jsonEvents;
+    private Long channelId;
+    private Long seqId;
+    private List<PoloniexWebSocketEvent> events;
 
-    public PoloniexWebSocketEvent[] getEvents() {
+    @JsonCreator
+    public PoloniexWebSocketEventsTransaction(
+            @JsonProperty("channelId") final Long channelId,
+            @JsonProperty("seqId") final Long seqId,
+            @JsonProperty("jsonEvents") final List<JsonNode> jsonEvents
+    ) {
+        this.channelId = channelId;
+        this.seqId = seqId;
+        this.events = createEvents(jsonEvents);
+    }
+
+    public Long getChannelId() {
+        return channelId;
+    }
+
+    public Long getSeqId() {
+        return seqId;
+    }
+
+    public List<PoloniexWebSocketEvent> getEvents() {
+        return events;
+    }
+
+    private List<PoloniexWebSocketEvent> createEvents(final List<JsonNode> jsonEvents) {
```

Comment:
Maybe not one for now, but I'm pretty sure all this work can be done by adding a Jackson deserializer annotation to `PoloniexWebSocketEvent` which causes the different event types to be deserialized as the concrete classes.

See the changes on https://github.com/bitrich-info/xchange-stream/pull/385/commits/7059acb767945daf35fce743ddc097e0b350438e to `LgoOrderEvent`

Reply:
I've originally tried to make the deserialization based purely on Jackson annotations, but had some problems, don't remember now, sorry.

Reply:
No problem. This is fine.